src/sframe/parallel_csv_parser.cpp

```diff
@@ -1017,6 +1018,13 @@ std::map<std::string, std::shared_ptr<sarray<flexible_type>>> parse_csvs_to_sfra
   file_and_status.clear(); // don't need these anymore
   
   // ensure that we actually found some valid files
+  if (files.empty() && !empty_files.empty()) {
+    // If all we got is empty files, let's take them --
+    // we'll produce an empty SFrame.
+    // But if we have any non-empty files, we should omit the
+    // empty ones so the column schema doesn't conflict.
+    files = empty_files;
```

Comment:
Or perhaps just exit here, and put back the lines below.
```
frame.open_for_write({},{},frame_sidx_file);
frame.close();
return {};
```

Reply:
Not sure if that would have quite the intended behavior -- I'm not sure what you mean by put back the lines below (I didn't remove those lines?). The way it's written currently, everything goes through the same path whether or not the files are empty (diverging only on whether they're all-empty or all-non-empty), and everything seems to work in both cases. I tested the following combinations:

* One empty file (results in empty SFrame with this change).
* One non-empty file (results in non-empty SFrame before and after).
* Two empty files (results in empty SFrame with this change).
* Two non-empty files (results in non-empty SFrame before and after).
* A mix of empty and non-empty files (with this change, ignores the empty files, same as before).

Reply:
Sorry, what I wrote was ambiguous.  Might this be a simpler alternative?

- Replacing this section with this:
```
frame.open_for_write({},{},frame_sidx_file);
frame.close();
return {};
```
 - restore the deletions at 1029-1031.

Reply:
Oh, I see. Sure, I'll make that change.